invenio_rdm_records/resources/deserializers/rocrate/__init__.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""RO-Crate deserializer."""
+
+from flask_resources import JSONDeserializer
+
+from .schema import ROCrateSchema
+
+
+class ROCrateJSONDeserializer(JSONDeserializer):
+    """RO-Crate metadata deserializer."""
+
+    def __init__(self, schema=ROCrateSchema):
+        """."""
+        self.schema = schema
+
+    def deserialize(self, data):
+        """Deserialize the RO-Crate payload."""
+        data = super().deserialize(data)
+        graph = data.get("@graph")
+        if not graph:
+            # TODO: Raise exception?
+            pass
```

Comment:
I agree on raising an exception and handling the error (e.g. return a HTTP 400)
Otherwise this will explode atm

Reply:
_plus one_ to this. We already have a [SerializerError](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/resources/serializers/errors.py#L11), we could do it in a similar fashion.

Alternative it can all be done _a la marshmallow_. `_dereference` could be a `pre_load` function. Then `graph` would be a top level -required- key. Throwing a `ValidationError` if not present.

Reply:
plus one to this